netsim/extra/repeater/defaults.yml

```diff
@@ -0,0 +1,9 @@
+# repeater plugin attributes and defaults
+#
+---
+attributes:
+  node:
+    repeat: int                     # Create N copies of this node anywhere it's used in groups or links
+
+repeater:
+  node_name_pattern: "{name[:13]}-{id:02x}"
```

Comment:
The generated node name is not an identifier, which obviously works for you because the code is executed in just the right place in the transformation process, but it might bite us in some unexpected places. It might be a good idea to replace "-" with "_"

Reply:
I specifically used '-' and not '_' because of some issues, see https://github.com/ipspace/netlab/pull/1619

'_' is not a valid hostname character

Note that '-' *is* valid in node ids, the logic used to verify them is at https://github.com/ipspace/netlab/blob/5dc32bf8ef219c650184be9ec652f5e13523fa23/netsim/data/types.py#L462 (and not https://github.com/ipspace/netlab/blob/5dc32bf8ef219c650184be9ec652f5e13523fa23/netsim/utils/strings.py#L196)

Reply:
How about we call ```make_id``` on whatever the (user provided) string template generates

Reply:
I'm also OK with '-' in the default template. As you said, it makes no sense to call **make_id** (also: defaults are not sanity-checked).